spartan/devnet-smoke-tests/deploy_token_contracts.sh

```diff
@@ -0,0 +1,36 @@
+token_contract_count=$(jq '.contracts | map(select(.type=="token")) | length' state.json)
+max_token_contracts=1
+token_contracts_per_run=1
+accounts=$(jq -c '.accounts[]' state.json)
+
+# Check the current token contract count and create more if current < max_token_contracts
+if [ "$token_contract_count" -lt "$max_token_contracts" ]; then
+  new_token_contracts="[]"
+
+  num_new_token_contracts=$(( token_contracts_per_run < (max_token_contracts - token_contract_count) ? token_contracts_per_run : (max_token_contracts - token_contract_count) ))
+  for i in $(seq 1 $num_new_token_contracts); do
+    admin_and_minter_address=$(select_random_account "$accounts")
+
+    token_symbol="TKN_$((token_contract_count + i))"
+    token_name="Token_$((token_contract_count + i))"
+
+    # We only use the prover on the first iteration of the loop to avoid duplicating identical proofs
```

Comment:
```suggestion
    # We only use the prover on the first iteration of the loop - while we need all tokens to be deployed, proving is
    # expensive and a single successful run is enough for the purposes of a smoke test.
```

(assuming this is what you meant)